### PR TITLE
[move-function] Add tests that show that the debugger displays variables correctly given usage of the move function.

### PR DESCRIPTION
**Pull request by gottesmm**
These are the first end to end tests that validate that showing moved variables
works correctly.

They work by just stepping through a program and validating the we can dump variables appropriately given their lifetimes.

### CODE REVIEW

**Comment by gottesmm** [comment]
Waiting for apple/swift#41523 to land.

**Comment by gottesmm** [comment]
@swift-ci test

File: lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+        # breakpoint.
+        self.runCmd('continue')
+
+    def do_check_copyable_var_test(self):
+        frame = self.thread.frames[0]
+        self.assertTrue(frame.IsValid(), "Couldn't get a frame.")
+
+        # We haven't defined varK yet.
+        varK = frame.FindVariable('k')
+        self.assertTrue(varK.value == None, "varK initialized too early?!")
```
Comment (kastiglione) [inline lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:119]: thank you for the tests!

Note there are many specific assertion functions, including `assertIsNone`, `assertEqual`, `assertGreater`, etc. They help produce better error messages if a test does fail.

Comment (gottesmm) [reply lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:119]: Ok. I'll send a follow on commit. I also noticed an fr v that snuck in there that is not necessary.

Comment (kastiglione) [reply lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:119]: I don't want to create work, I mean only to mention it for future.

Comment (gottesmm) [reply lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:119]: @kastiglione I had other things I needed to do and it inspired me to port some tests.

File: lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+        self.target = self.dbg.CreateTarget(self.exec_artifact)
+        self.assertTrue(self.target, VALID_TARGET)
+
+        self.do_setup_breakpoints()
+
+        self.process = self.target.LaunchSimple(None, None, os.getcwd())
+        threads = lldbutil.get_threads_stopped_at_breakpoint(
+            self.process, self.breakpoints[0])
+        self.assertTrue(len(threads) == 1)
+        self.thread = threads[0]
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:49]: You might be able to achieve the same with 
```
        target, process, thread, bkpt = lldbutil.run_to_source_breakpoint(
            self, 'stop here', lldb.SBFileSpec('main.swift'))
       self.assertGreater(bkpt.GetNumLocations(), 6)
```

File: lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+        self.runCmd('continue')
+        self.assertTrue(varK.unsigned > 0, "varK not initialized?!")
+
+        # Go to breakpoint 3. We invalidated k
+        self.runCmd('continue')
+        self.assertTrue(varK.value == None, "K is live but was moved?!")
+
+        # Go to the last breakpoint and make sure that k is reinitialized
+        # properly.
+        self.runCmd('continue')
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:131]: if you store the process above, you could do `process.Continue()` here, which is slightly nicer than parsing a string command.

File: lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+
+        self.assertTrue(varK.value == None, "varK initialized too early?!")
+
+        # Go to break point 2. k should be valid.
+        self.runCmd('continue')
+        self.assertTrue(varK.unsigned > 0, "varK not initialized?!")
+
+        # Go to breakpoint 3. k should no longer be valid.
+        self.runCmd('continue')
+        self.assertTrue(varK.value == None, "K is live but was moved?!")
```
Comment (adrian-prantl) [inline lldb/test/API/lang/swift/variables/move_function/TestSwiftMoveFunction.py:107]: `self.assertEqual()` will print the value the the comparison fails for easier debugging.

see `lldb/third_party/Python/module/unittest2/unittest2/case.py` for a full list

**Pull request merged**